File: ds/array/reverseAString/reverseAString.js
function reverse(str) {
    // check input
    if (!str || str.length < 2 || typeof str !== 'string') {
        return 'need a valid string'
    }

    let ans = '';

    const totalItems = str.length - 1;
    for (let i = totalItems; i >= 0; i--) {
        ans += str[i]
    }

    return ans;
}

module.exports = reverse;

File: ds/array/reverseAString/reverseAString.py
def reverse(s):
    # check input
    if (not s or type(s) != str or len(s) < 2):
        return 'need a valid string'
    
    ans = ''
    
    totalItems = len(s) - 1
    for i in range(totalItems, -1, -1):
        ans += s[i]
    
    return ans
